### PR TITLE
Add monkeypatching to the `mol_data` tests

### PR DESCRIPTION
**Pull request by obliviateandsurrender**
**Context:**
Sometimes `mol_data` tests were failing due to the PubChem server being busy.

**Description of the Change:**
Moneypatched the methods `pubchem.get_cids` and `pubchem.from_cid` that relied on pinging PubChem servers for data.

**Benefits:**
Tests will no longer rely upon pinging the PubChem servers.

### CODE REVIEW

File: tests/qchem/test_structure.py [hunk 1]
```diff
@@ -314,6 +317,61 @@ def test_inconsistent_active_spaces(
         )
 
 
+def mock_get_cids(identifier, namespcae, pcp=None):
```
Comment (soranjh) [inline tests/qchem/test_structure.py:320]: Why we need this function? If I understand `mol_data` correctly, the data is accessed by defining an identifier. Why there is a need to return the CID?

File: tests/qchem/test_structure.py [hunk 2]
```diff
@@ -314,6 +317,61 @@ def test_inconsistent_active_spaces(
...
+    """Return PubChem Compound ID for the provided identifer"""
```
Comment (soranjh) [inline tests/qchem/test_structure.py:321]:
```suggestion
    """Return PubChem Compound ID for the provided identifier"""
```

File: tests/qchem/test_structure.py [hunk 3]
```diff
@@ -314,6 +317,61 @@ def test_inconsistent_active_spaces(
...
+        297: {
+            "atoms": {"aid": [1, 2, 3, 4, 5], "element": [6, 1, 1, 1, 1]},
+            "coords": [
+                {
+                    "aid": [1, 2, 3, 4, 5],
+                    "conformers": [
+                        {
+                            "x": [0, 0.5541, 0.6833, -0.7782, -0.4593],
+                            "y": [0, 0.7996, -0.8134, -0.3735, 0.3874],
+                            "z": [0, 0.4965, -0.2536, 0.6692, -0.9121],
```
Comment (soranjh) [inline tests/qchem/test_structure.py:355]: Why PubChem has 5 xyz elements?

File: tests/qchem/test_structure.py [hunk 4]
```diff
@@ -314,6 +317,61 @@ def test_inconsistent_active_spaces(
...
+                        }
+                    ],
+                }
+            ],
+        },
+        783: {
+            "atoms": {"aid": [1, 2], "element": [1, 1]},
+            "coords": [{"aid": [1, 2], "conformers": [{"x": [2, 3], "y": [0, 0]}]}],
```
Comment (soranjh) [inline tests/qchem/test_structure.py:363]: What does it mean? `[{"x": [2, 3], "y": [0, 0]}]`

File: tests/qchem/test_structure.py [hunk 5]
```diff
@@ -314,6 +317,61 @@ def test_inconsistent_active_spaces(
...
+        },
+    }
+
+    if [cid, record_type] in [[297, "3d"], [783, "2d"]]:
+        return pcp.Compound(records[cid])
+
+    raise pcp.NotFoundError if [cid, record_type] == [783, "3d"] else ValueError(
+        "Provided CID (or Identifier) is None."
```
Comment (soranjh) [inline tests/qchem/test_structure.py:371]: Does it exactly replicate PubChem behaviour?

**Review by soranjh — CHANGES_REQUESTED** [review]

File: tests/qchem/test_structure.py (hunk 1 above)
Comment (obliviateandsurrender) [reply tests/qchem/test_structure.py:320]: The workflow in `mol_data` is to first obtain the CID for the given molecule using any non-CID identifier they provide via `pubchempy.get_cids` method and then using the obtained CID for obtaining the `pubchem.Compound` data via `pubchem.from_cid`. So I am mocking both these methods here.

Comment (soranjh) [reply tests/qchem/test_structure.py:320]: `pubchempy` is only needed to get the CID from on the identifiers?

**Review by soranjh — APPROVED** [review]

**Review by soranjh — APPROVED** [review]
Please update the changelog.

File: tests/qchem/test_structure.py (hunk 1 above)
Comment (obliviateandsurrender) [reply tests/qchem/test_structure.py:320]: Also, to get the molecular data using that retrieved CID.

**Pull request merged**